common/scanner.h

```diff
@@ -114,6 +116,20 @@ typedef struct {
 
 typedef enum { Error, End } ScanContentResult;
 
+static inline signed string_cmp(String a, String b) {
+    if (a.len != b.len) {
+        return a.len < b.len ? -1 : 1;
+    }
+
+    for (uint32_t i = 0; i < a.len; i++) {
+        if (a.data[i] != b.data[i]) {
+            return a.data[i] < b.data[i] ? -1 : 1;
+        }
+    }
+
+    return 0;
+}
```

Comment:
hm, any reason to prefer this over strcmp?

Reply:
This compares two Strings regardless of the `data` datatype (no longer a char). I couldn't get `strcmp` to work but I could be doing something wrong:

The new test passes:

```
<<<漢漢字
  This is a heredoc
漢漢字;
```

but this one fails:

```
<<< EOF
E
EOF;
```

![image](https://github.com/tree-sitter/tree-sitter-php/assets/4176520/5c4ead1a-0e67-410a-820e-cd165093601b)

Reply:
wcscmp?

Reply:
Ah sweet thanks! I'm not as familiar with all the c libs

Reply:
no worries, there's too much in them anyways :laughing: